### ISSUE TITLE
Role attribute is ignored in the provider if the corresponding "SNOWFLAKE_ROLE" env variable is set

### ISSUE THREAD

**Issue by pashamartinenko** [Snowflake-Labs/terraform-provider-snowflake#2242]
### Terraform CLI and Provider Versions

Terraform v1.3.7
on linux_amd64

provider registry.terraform.io/snowflake-labs/snowflake snowflake-labs/snowflake v0.74.0

### Terraform Configuration

```terraform
provider "snowflake" {
  // account - pulled from env variable
  // username - pulled from env variable
  // password - pulled from env variable
  // role - pulled from env variable (SNOWFLAKE_ROLE is set to ROLE_1)
}

provider "snowflake" {
  // account - pulled from env variable
  // username - pulled from env variable
  // password - pulled from env variable
  role  = "ROLE_2"
  alias = "role2"
}

resource "snowflake_database" "database1" {
  name                        = "database_name1"
}

resource "snowflake_database" "database2" {
  name                        = "database_name2"
  provider = snowflake.role2
}
```

### Expected Behavior

`database_name1` db owner is `ROLE_1`, `database_name2` owner is `ROLE_2`

### Actual Behavior

both `database_name1` and `database_name2`  dbs' owner is `ROLE_1`

### Steps to Reproduce

1. run terraform init, plan, apply in a new workspace using v0.73.0 provider - returns expected result
1. run terraform init, plan, apply in a new workspace using v0.74.0 provider - produces the described issue

### How much impact is this issue causing?

Low

### Additional Information

workaround is unsetting SNOWFLAKE_ROLE env variable and assigning `role` attribute to both providers explicitly

**Comment by AZenat** [comment]
Same issue with v0.86.
Which is weird because according to [the code](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/f3c85c0ebe3e54dc91836fe998c2d7c8e3373a52/framework/provider/provider.go#L442), the value written in the provider should override the environment variable.

**Comment by sfc-gh-asawicki** [comment]
Hey @AZenat @PedroMartinSteenstrup @pashamartinenko. This seems connected to #2294. We plan to rework the configuration completely but because this is really misleading now and can result in unpredicted behavior, we will adjust the current config behavior to what the docs say.

I hope the fix will land next week (0.87.0 or 0.88.0).

**Comment by sfc-gh-asawicki** [comment]
Closing the issue as it was solved and confirmed by @PedroMartinSteenstrup.

@pashamartinenko please open a new one if you still encounter the problems using the newest provider version.
